Mathlib/NumberTheory/NumberField/CMField.Lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2025 X. Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import Mathlib.NumberTheory.NumberField.Units.DirichletTheorem
+
+/-!
+# CM extension of number fields
+
+A CM-extension of number fields is a totally complex number field that is a quadratic extension of
+a totally real number field.
+
+## Main definitions
+
+* `IsCM`: a predicate on two number fields `F` and `K` stating that `K/F` is a CM-extension.
+
+-/
+
+open NumberField InfinitePlace ComplexEmbedding
+
+namespace NumberField
+
+/--
+An extension `K/F` of number fields is a `CM`-extension if `K` is totally complex, `F` is totally
+real and the extension `K/F` is of degree `2`.
+-/
+class IsCM (F K : Type*) [Field F] [NumberField F] [Field K] [NumberField K]
```

Comment:
I confess I am not entirely sure of this definition; why do you want to fix the totally real subfield as a variable?

Reply:
What would be the other possibility? Note that I define the `maximalRealSubfield` of a totally complex field in #23760 and prove that, if `K/F` is CM then `F ≃ₐ[ℚ] maximalRealSubfield ℚ K` in a coming PR.